File: contactsinroomlibraryjava/src/main/java/com/example/contactsinroomlibraryjava/MainActivity.java
package com.example.contactsinroomlibraryjava;


import androidx.appcompat.app.AlertDialog;
import androidx.appcompat.app.AppCompatActivity;

import androidx.recyclerview.widget.DefaultItemAnimator;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;
import androidx.room.Room;

import android.content.DialogInterface;
import android.os.Bundle;
import android.text.TextUtils;
import android.view.LayoutInflater;
import android.view.View;
import android.widget.CheckBox;
import android.widget.EditText;
import android.widget.ImageButton;
import android.widget.ImageView;
import android.widget.TextView;
import android.widget.Toast;

import com.example.contactsinroomlibraryjava.data.ContactAppDatabase;
import com.example.contactsinroomlibraryjava.recyclerViewAdapter.AdapterContacts;

import java.util.ArrayList;

import static com.example.contactsinroomlibraryjava.utils.Util.DATABASE_NAME;

public class MainActivity extends AppCompatActivity {
    private long id;
    private AdapterContacts adapterContacts;
    private ArrayList<Contact> arrayListContacts = new ArrayList<>();
    private RecyclerView recyclerView;
    private ImageButton imageButton;
    private EditText nameEditText;
    private TextView newContactTitle;
    private EditText informationEditText;
    private AlertDialog alertDialog;
    private View view;
    private AlertDialog.Builder alertDialogBuilderUserInput;
    private LayoutInflater layoutInflaterAndroid;
    private RecyclerView.LayoutManager mLayoutManager;
    private Contact contact;
    private ContactAppDatabase contactAppDatabase;
    private CheckBox checkBoxPhone, checkBoxEmail;
    private ImageView imageViewPhoto;
    private boolean checkedCheckBox;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        recyclerView = findViewById(R.id.recyclerView);
        contactAppDatabase = Room.databaseBuilder(getApplicationContext(),
                ContactAppDatabase.class,
                DATABASE_NAME).allowMainThreadQueries().build();
        arrayListContacts.addAll(contactAppDatabase.getContactDAO().getAllContacts());
        adapterContacts = new AdapterContacts(this, arrayListContacts, MainActivity.this);
        mLayoutManager = new LinearLayoutManager(getApplicationContext());
        recyclerView.setLayoutManager(mLayoutManager);
        recyclerView.setItemAnimator(new DefaultItemAnimator());
        recyclerView.setAdapter(adapterContacts);
        imageViewPhoto = findViewById(R.id.photoContact);
        imageButton = findViewById(R.id.imageButtonAdd);
        imageButton.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                addAndEditContact(false, null, -1);
            }
        });
    }

    public void addAndEditContact(final boolean isUpdate, final Contact contact, final int position) {

        layoutInflaterAndroid = LayoutInflater.from(getApplicationContext());
        view = layoutInflaterAndroid.inflate(R.layout.activity_add_contacts, null);
        alertDialogBuilderUserInput = new AlertDialog.Builder(MainActivity.this);
        alertDialogBuilderUserInput.setView(view);
        newContactTitle = view.findViewById(R.id.newContactTitle);
        nameEditText = view.findViewById(R.id.nameEditText);
        checkBoxPhone = view.findViewById(R.id.checkBoxPhone);
        checkBoxEmail = view.findViewById(R.id.checkBoxEmail);
        informationEditText = view.findViewById(R.id.informationEditText);
        checkingTwoParameters();
        newContactTitle.setText(!isUpdate ? "Добавить контакт" : "Изменение контакта");
        if (isUpdate && contact != null) {
            nameEditText.setText(contact.getName());
            informationEditText.setText(contact.getInformation());
        }
        alertDialogBuilderUserInput
                .setCancelable(false)
                .setPositiveButton(isUpdate ? "Обновить" : "Сохранить", new DialogInterface.OnClickListener() {
                    public void onClick(DialogInterface dialogBox, int id) {

                    }
                })
                .setNegativeButton(isUpdate ? "Удалить" : "Отмена", new DialogInterface.OnClickListener() {
                    public void onClick(DialogInterface dialogBox, int id) {

                        if (isUpdate) {
                            deleteContact(contact, position);
                        } else {
                            dialogBox.cancel();
                        }

                    }
                });

        alertDialog = alertDialogBuilderUserInput.create();
        alertDialog.show();
        alertDialog.getButton(AlertDialog.BUTTON_POSITIVE).setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {

                if (TextUtils.isEmpty(nameEditText.getText().toString())) {
                    Toast.makeText(MainActivity.this, "Введите имя!", Toast.LENGTH_SHORT).show();
                    return;
                } else if (TextUtils.isEmpty(informationEditText.getText().toString())) {
                    Toast.makeText(MainActivity.this, "Введите информацию о пользователе", Toast.LENGTH_SHORT).show();
                    return;
                } else {
                    alertDialog.dismiss();
                }
                if (isUpdate && contact != null) {

                    updateContact(nameEditText.getText().toString(), informationEditText.getText().toString(), position);
                } else {
                    createContact(nameEditText.getText().toString(), informationEditText.getText().toString());
                }
            }
        });
    }

    private void checkingTwoParameters() {
        checkBoxEmail.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                if (checkBoxEmail.isChecked()) {
                    checkedCheckBox = true;
                    Toast.makeText(MainActivity.this, "Введите свою почту!", Toast.LENGTH_SHORT).show();
                }
            }
        });
        checkBoxPhone.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                if (checkBoxPhone.isChecked()) {
                    checkedCheckBox = false;
                    Toast.makeText(MainActivity.this, "Введите свой телефон", Toast.LENGTH_SHORT).show();
                }
            }
        });
    }

    private void deleteContact(Contact contact, int position) {
        arrayListContacts.remove(position);
        contactAppDatabase.getContactDAO().deleteContact(contact);
        adapterContacts.notifyDataSetChanged();
    }

    private void updateContact(String name, String info, int position) {
        contact = arrayListContacts.get(position);
        contact.setName(name);
        contact.setInformation(info);
        if (checkedCheckBox) {
            contact.setPhoto(R.drawable.ic_baseline_email_turquoise);
        } else {
            contact.setPhoto(R.drawable.ic_baseline_contact_phone_purple);
        }
        contactAppDatabase.getContactDAO().updateContact(contact);
        arrayListContacts.set(position, contact);
        adapterContacts.notifyDataSetChanged();
    }

    private void createContact(String name, String info) {
        if (checkedCheckBox) {
            id = contactAppDatabase.getContactDAO().addContact(new Contact(0, name, info, R.drawable.ic_baseline_email_turquoise,1));
        } else {
            id = contactAppDatabase.getContactDAO().addContact(new Contact(0, name, info, R.drawable.ic_baseline_contact_phone_purple,2));
        }
        contact = contactAppDatabase.getContactDAO().getContact(id);
        if (contact != null) {
            arrayListContacts.add(0, contact);
            adapterContacts.notifyDataSetChanged();
        }
    }
}

File: contactsinroomlibrary/src/main/java/com/example/contactsinroomlibrary/MainActivity.kt
package com.example.contactsinroomlibrary

import android.os.Bundle
import android.text.TextUtils
import android.view.LayoutInflater
import android.view.View
import android.widget.Toast
import androidx.appcompat.app.AlertDialog
import androidx.appcompat.app.AppCompatActivity
import androidx.recyclerview.widget.DefaultItemAnimator
import androidx.recyclerview.widget.LinearLayoutManager
import androidx.room.Room
import com.example.contactsinroomlibrary.data.ContactAppDatabase
import com.example.contactsinroomlibrary.recyclerviewadapter.AdapterContacts
import kotlinx.android.synthetic.main.activity_add_contacts.view.*
import kotlinx.android.synthetic.main.activity_main.*
import java.util.ArrayList

class MainActivity : AppCompatActivity() {
    private var id: Long = 0
    private lateinit var adapterContacts: AdapterContacts
    private val arrayListContacts = ArrayList<Contact>()
    private lateinit var view: View
    private lateinit var contact: Contact
    private lateinit var contactAppDatabase: ContactAppDatabase
    private var checkedCheckBox = false

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        val mLayoutManager = LinearLayoutManager(applicationContext)
        contactAppDatabase = Room.databaseBuilder(applicationContext, ContactAppDatabase::class.java, "DB").allowMainThreadQueries().build()
        contactAppDatabase.getContactDAO().getAllContacts().let { arrayListContacts.addAll(it) }
        adapterContacts = AdapterContacts(this, arrayListContacts, this@MainActivity)
        recyclerView.layoutManager = mLayoutManager
        recyclerView.itemAnimator = DefaultItemAnimator()
        recyclerView.adapter = adapterContacts
        imageButtonAdd.setOnClickListener { addAndEditContact(false, null, -1) }
    }

    fun addAndEditContact(isUpdate: Boolean, contact: Contact?, position: Int) { // private не ставлю так как вызываю эту функцию в AdapterContact
        val layoutInflaterAndroid = LayoutInflater.from(applicationContext)
        view = layoutInflaterAndroid.inflate(R.layout.activity_add_contacts, null)
        val alertDialogBuilderUserInput = AlertDialog.Builder(this@MainActivity)
        alertDialogBuilderUserInput.setView(view)
        checkingTwoParameters()
        view.newContactTitle.text = if (!isUpdate) "Добавить контакт" else "Изменение контакта"
        if (isUpdate && contact != null) {
            view.nameEditText.run { setText(contact.name) }
            view.informationEditText.setText(contact.information)
        }
        alertDialogBuilderUserInput
                .setCancelable(false)
                .setPositiveButton(if (isUpdate) "Обновить" else "Сохранить") { _, _ -> }
                .setNegativeButton(if (isUpdate) "Удалить" else "Отмена") { dialogBox, _ ->
                    if (isUpdate) {
                        deleteContact(contact, position)
                    } else {
                        dialogBox.cancel()
                    }
                }
        val alertDialog = alertDialogBuilderUserInput.create()
        alertDialog.show()
        alertDialog.getButton(AlertDialog.BUTTON_POSITIVE).setOnClickListener(View.OnClickListener {
            when {
                TextUtils.isEmpty(view.nameEditText.text.toString()) -> {
                    Toast.makeText(this@MainActivity, "Введите имя!", Toast.LENGTH_SHORT).show()
                    return@OnClickListener
                }
                TextUtils.isEmpty(view.informationEditText.text.toString()) -> {
                    Toast.makeText(this@MainActivity, "Введите информацию о пользователе", Toast.LENGTH_SHORT).show()
                    return@OnClickListener
                }
                else -> {
                    alertDialog.dismiss()
                }
            }
            if (isUpdate && contact != null) {
                updateContact(view.nameEditText.text.toString(), view.informationEditText.text.toString(), position)
            } else {
                createContact(view.nameEditText.text.toString(), view.informationEditText.text.toString())
            }
        })
    }

    private fun checkingTwoParameters() {
        view.checkBoxEmail.setOnClickListener {
            if (view.checkBoxEmail.isChecked) {
                checkedCheckBox = true
                Toast.makeText(this@MainActivity, "Введите свою почту!", Toast.LENGTH_SHORT).show()
            }
        }
        view.checkBoxPhone.setOnClickListener {
            if (view.checkBoxPhone.isChecked) {
                checkedCheckBox = false
                Toast.makeText(this@MainActivity, "Введите свой телефон", Toast.LENGTH_SHORT).show()
            }
        }
    }

    private fun deleteContact(contact: Contact?, position: Int) {
        arrayListContacts.removeAt(position)
        contact?.let { contactAppDatabase.getContactDAO().deleteContact(it) }
        adapterContacts.notifyDataSetChanged()
    }

    private fun updateContact(name: String, info: String, position: Int) {
        contact = arrayListContacts[position]
        contact.name = name
        contact.information = info
        if (checkedCheckBox) {
            contact.photo = R.drawable.ic_baseline_email_turquoise
        } else {
            contact.photo = R.drawable.ic_baseline_contact_phone_purple
        }
        contactAppDatabase.getContactDAO().updateContact(contact)
        arrayListContacts[position] = contact
        adapterContacts.notifyDataSetChanged()
    }

    private fun createContact(name: String, info: String) {
        id = if (checkedCheckBox) {
            contactAppDatabase.getContactDAO().addContact(Contact(0, name, info, R.drawable.ic_baseline_email_turquoise))
        } else {
            contactAppDatabase.getContactDAO().addContact(Contact(0, name, info, R.drawable.ic_baseline_contact_phone_purple))
        }
        contact = contactAppDatabase.getContactDAO().getContact(id)
        arrayListContacts.add(0, contact)
        adapterContacts.notifyDataSetChanged()
    }
}